### PR TITLE
refactor(ram_monitor): ros2 port

### PR DESCRIPTION
**Pull request by reinzor**

### CODE REVIEW

**Comment by reinzor** [comment]
ping @ct2034

**Comment by ct2034** [comment]
Tested locally ✅

File: diagnostic_common_diagnostics/README.md [hunk 1]
```diff
@@ -70,7 +70,27 @@ Disable self test.
...
 
 ## ram_monitor.py
-**To be ported**
+The `ram_monitor` module allows users to monitor the RAM usage of their system in real-time.
+It publishes the usage percentage in a diagnostic message.
+
+* Name of the node is "ram_monitor_" + hostname.
+* Uses the following args:
+  * warning_percentage: If the RAM usage is > warning_percentage, a WARN status will be published.
+  * window: the maximum length of the used collections.deque for queuing CPU readings.
```
Comment (ct2034) [inline diagnostic_common_diagnostics/README.md]:
```suggestion
  * window: the maximum length of the used collections.deque for queuing RAM readings.
```
(I guess)

File: diagnostic_common_diagnostics/diagnostic_common_diagnostics/ram_monitor.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+def main():
+    hostname = socket.gethostname()
+    rclpy.init()
+    node = rclpy.create_node('ram_monitor_%s' % hostname.replace('-', '_'))
+
+    updater = Updater(node)
+    updater.setHardwareID(hostname)
+    updater.add(
+        RamTask(
+            node.declare_parameter('warning_percentage', 90.0).value,
```
Comment (ct2034) [inline diagnostic_common_diagnostics/diagnostic_common_diagnostics/ram_monitor.py]:
```suggestion
            node.declare_parameter('warning_percentage', 90).value,
```
Why set it as float here, then cast it to an int?

File: diagnostic_common_diagnostics/diagnostic_common_diagnostics/ram_monitor.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+from diagnostic_updater import DiagnosticTask, Updater
+
+import psutil
+
+import rclpy
+
+
+class RamTask(DiagnosticTask):
+
+    def __init__(self, warning_percentage, window=1):
```
Comment (ct2034) [inline diagnostic_common_diagnostics/diagnostic_common_diagnostics/ram_monitor.py]:
```suggestion
    def __init__(self, warning_percentage, window):
```
I think default value makes more sense on ROS level

File: diagnostic_common_diagnostics/diagnostic_common_diagnostics/ram_monitor.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        DiagnosticTask.__init__(self, 'RAM Information')
+        self._warning_percentage = int(warning_percentage)
+        self._readings = collections.deque(maxlen=window)
+
+    def run(self, stat):
+        self._readings.append(psutil.virtual_memory().percent)
+        ram_average = sum(self._readings) / len(self._readings)
+
+        stat.add('RAM Load Average', '{:.2f}'.format(ram_average))
```
Comment (ct2034) [inline diagnostic_common_diagnostics/diagnostic_common_diagnostics/ram_monitor.py]: Please use f-strings everywhere

**Review by ct2034 — CHANGES_REQUESTED** [review]

**Comment by reinzor** [comment]
@ct2034 , can we merge this?

**Comment by reinzor** [comment]
ping @ct2034

**Pull request merged**